vdbench.py

```diff
@@ -1,367 +1,367 @@
-File Name	:	vdbench.py
-Description	:	This script used to execute vdbench tool and create
-                HTML reports.
-
-"""
-import os
-import sys
-from configparser import ConfigParser
-import subprocess
-path_ = os.path.abspath("../../../Interface/REST")
-sys.path.insert(0, path_)
-from ILDC import ILDC
-sys.path.insert(0, os.path.abspath("../../../Lib/VdBenchLib"))
-from error_log import LogCreat
-
-class VdBenchRun():
-    '''
-    Class:- VdBenchRun
-    This class executes vdbench 4k fill and VSI/VDI/ORACLE/SQL
-    wokload as user request.
-    Arguments : None
-    Return: None
-    '''
-    def __init__(self):
-        configur = ConfigParser()
-        configur.read(r"../../../Config/VdBench_config/VDBench_config.ini")
-        self.vdbench_path = configur.get('Vdbench run', 'vdbench_executable_path')
-        self.time_stamp = configur.get('first run', 'start')
-        self.build = ''
-        self.new_ = ''
-    def run(self,vd_name, workload, diskindex):
-        '''
-        This method execute workload of VdBench tool.
-        Arguments (str, str, int): vd_name, workload,diskindex
-        Return: None
-        '''
-        workload_path, result_path = self.create_file('4-4k-4-fill.vdb', vd_name,diskindex, workload)
-        self.run_workload(workload_path, result_path,vd_name,workload)
-        if workload.strip() == 'VDI':
-            workload_path, result_path = self.create_file('vdi_fill.vdb', vd_name,diskindex,workload)
-        elif workload.strip() == 'VSI':
-            workload_path, result_path = self.create_file('vsi_fill.vdb', vd_name,diskindex,workload)
-        elif workload.strip() == 'ORACLE':
-            workload_path, result_path = self.create_file('oracle_fill.vdb', vd_name,diskindex,workload)
-        else:
-            workload_path, result_path = self.create_file('sql_fill.vdb', vd_name,diskindex,workload)
-        self.run_workload(workload_path, result_path,vd_name, workload)
-
-    def run_workload(self, workload_path, result_path, vd_name, workload):
-        '''
-        This method run workloads and stores the result
-        Arguments (str, str, str, str): workload_path, result_path, vd_name, workload
-        Return: None
-        '''
-        # Config\VdBench_config\Workload
-        path = 'cd ' + self.vdbench_path +'\n'
-        str_ = 'vdbench -f "' + workload_path + '" -o "'+result_path+'"'+"\n"
-        ssh = subprocess.Popen(["cmd"],
-                            stdin =subprocess.PIPE,
-                            stdout=subprocess.PIPE,
-                            stderr=subprocess.PIPE,
-                            universal_newlines=True,
-                            bufsize=0)
-        # Send ssh commands to stdin
-        ssh.stdin.write(path)
-        ssh.stdin.write(str_)
-        out, err = ssh.communicate()
-        print(out,err)
-        ssh.wait()
-        if 'Vdbench execution completed successfully' in out:
```

Comment:
Is there no other way to check if vdbench execution is successfull other thatn string comparision ?.